### ISSUE TITLE
Hook conflicting with npm version

### ISSUE THREAD

**Issue by sofianegargouri** [carloscuesta/gitmoji-cli#497]
Hello @carloscuesta!

Issue:

I'm starting to use yarn / npm versions to automagically update my package version and publish a tag on Git. However, the gitmoji hook prevents me to do so without having to `Ctrl + C` multiple times.

Example:
```
➜ yarn version --minor
yarn version v1.22.4
info Current version: 0.2.0
info New version: 0.3.0

*Nothing happens until I press Ctrl+C*

$ react-native-version
[RNV] Versioning Android...
[RNV] Android updated
[RNV] Versioning iOS...
[RNV] iOS updated
[RNV] Amending...

*Nothing happens until I press Ctrl+C*

? Choose a gitmoji: (Use arrow keys or type to search)
❯ 🎨  - Improve structure / format of the code. 
  ⚡️  - Improve performance. 
  🔥  - Remove code or files. 
  🐛  - Fix a bug. 
  🚑  - Critical hotfix. 
  ✨  - Introduce new features. 
  📝  - Add or update documentation. 
(Move up and down to reveal more choices)
[RNV] Adjusting Git tag...
[RNV] Done
✨  Done in 33.40s.
```

This only happens when the gitmoji hook is enabled. If I remove it, everything works flawlessly

**OS**: Mac OS Big Sur
**gitmoji -v**: 3.2.12
**Node version**: Irrelevant, tried on multiple node versions (v10 and v13)

**Comment by carloscuesta** [comment]
I think we should implement something similar to https://github.com/carloscuesta/gitmoji-cli/pull/445

Seems that the hook is running up on the `Ammending` step and that's why you need to cancel it manually

**Comment by Jeremie-Chauvel** [comment]
Hi looking into this,
for strategy, I will be using the second argument passed to the hook `prepare-commit-msg` to detect when `Ammending` or reusing the same commit message: https://git-scm.com/docs/githooks#_prepare_commit_msg.

**Comment by Jeremie-Chauvel** [comment]
Hi @carloscuesta, 

sadly yarn version is creating a commit using `git commit -m` and cancelling the commit message for this case would mean canceling th commit message whenever the message is already defined https://git-scm.com/docs/githooks#_prepare_commit_msg and would means discarding https://github.com/carloscuesta/gitmoji-cli/pull/446 which is pretty neat.

I'm submitting my code to abord the coomit hook when amending, I can let you decide if the feature is interesting.

## TLDR
- `yarn version` can't be fixed on gitmoji-cli side without breaking functionality 
- @sofianegargouri  you can fix the hang on yarn side by using the config `yarn config set version-commit-hooks false` at your own risk https://classic.yarnpkg.com/en/docs/cli/version/#toc-git-tags

**Comment by Jeremie-Chauvel** [comment]
Another lead would be to look into `tty` detection but again would be a risky change
